db/migrations/20230406025628_create_user_restriction_table.js

```diff
@@ -0,0 +1,18 @@
+const { baseDown } = require('../utils')
+
+const table = 'user_restriction'
+
+exports.up = async (knex) => {
+  await knex('entity_type').insert({ table })
+  await knex.schema.createTable(table, (t) => {
+    t.bigIncrements('id').primary()
+    t.bigInteger('user_id').unsigned().notNullable()
+    t.enu('type', ['articleHottest', 'articleNewest']).notNullable()
+    t.timestamp('created_at').defaultTo(knex.fn.now())
+
+    t.index(['user_id'])
```

Comment:
also index `type`?

Reply:
enum column may not benefit too much from indexing ?

Reply:
we use filtering by `type` in [recommendation resolvers](https://github.com/thematters/matters-server/pull/3273/files#diff-f090852af5161630e96eabe822e33d6c3dc8aefcc8812c00807235f6a39a4728R45-R50), any performance impact?

Reply:
Indexed enum column first scan index and read, let say 30%, rows row by row from disk,  comparing to batch read all rows from disk in the un-indexed case,   it hard to say which one can be faster.